src/lazy/text/buffer.rs

```diff
@@ -967,6 +978,208 @@ impl<'data> TextBufferView<'data> {
+                Self::match_timestamp_month,
+                Self::match_timestamp_day,
+                Self::match_timestamp_hour_and_minute,
+                Self::match_timestamp_seconds,
+                Self::match_timestamp_fractional_seconds,
+                Self::match_timestamp_offset,
+            )),
+            Self::peek_stop_character,
+        )
+        .map(|(_y, _m, _d, _hm, _s, _f, offset)| {
+            MatchedTimestamp::new(TimestampPrecision::Second).with_offset(offset)
+        })
+        .parse(self)
+    }
+
+    /// Matches the year component of a timestamp.
+    fn match_timestamp_year(self) -> IonMatchResult<'data> {
+        recognize(take_while_m_n(4, 4, |c: u8| c.is_ascii_digit()))(self)
+    }
+
+    /// Matches the month component of a timestamp, including a leading `-`.
+    fn match_timestamp_month(self) -> IonMatchResult<'data> {
+        preceded(
+            tag("-"),
+            recognize(alt((
+                pair(char('0'), one_of("123456789")),
+                pair(char('1'), one_of("012")),
+            ))),
+        )(self)
+    }
+
+    /// Matches the day component of a timestamp, including a leading `-`.
+    fn match_timestamp_day(self) -> IonMatchResult<'data> {
+        preceded(
+            tag("-"),
+            recognize(alt((
+                pair(char('0'), one_of("123456789")),
+                pair(one_of("12"), Self::match_any_digit),
+                pair(char('3'), one_of("10")),
+            ))),
+        )(self)
+    }
+
+    /// Matches a leading `T`, a two-digit hour component of a timestamp, a delimiting ':', and a
+    /// two-digit minute component.
+    fn match_timestamp_hour_and_minute(
+        self,
+    ) -> IonParseResult<'data, (TextBufferView<'data>, TextBufferView<'data>)> {
+        preceded(
+            tag("T"),
+            // Hour
+            separated_pair(
+                recognize(alt((
+                    pair(one_of("01"), Self::match_any_digit),
+                    pair(char('2'), one_of("0123")),
+                ))),
+                // Delimiter
+                tag(":"),
+                // Minutes
+                recognize(pair(one_of("012345"), Self::match_any_digit)),
+            ),
+        )(self)
+    }
+
+    /// Matches a leading `:`, and any two-digit second component from `00` to `59` inclusive.
+    fn match_timestamp_seconds(self) -> IonMatchResult<'data> {
+        preceded(
+            tag(":"),
+            recognize(pair(one_of("012345"), Self::match_any_digit)),
+        )(self)
+    }
+
+    /// Matches the fractional seconds component of a timestamp, including a leading `.`.
+    fn match_timestamp_fractional_seconds(self) -> IonMatchResult<'data> {
+        preceded(tag("."), digit1)(self)
+    }
+
+    /// Matches a timestamp offset of any format.
+    fn match_timestamp_offset(self) -> IonParseResult<'data, MatchedTimestampOffset> {
+        alt((
+            value(MatchedTimestampOffset::Zulu, tag("Z")),
```

Comment:
It might be a wash; it makes reading `+00:00` cheaper at the expense of an extra branch in the general case. I've added the change for now, we can always revisit it later with benchmarks.